policy-controller/k8s/index/src/http_route.rs

```diff
@@ -265,4 +219,138 @@ impl InboundRouteBinding {
         };
         Ok(filter)
     }
+
+    fn try_policy_filter(filter: policy::HttpRouteFilter) -> Result<http_route::InboundFilter> {
+        let filter = match filter {
+            policy::HttpRouteFilter::RequestHeaderModifier {
+                request_header_modifier,
+            } => {
+                let filter = convert::req_header_modifier(request_header_modifier)?;
+                http_route::InboundFilter::RequestHeaderModifier(filter)
+            }
+
+            policy::HttpRouteFilter::RequestRedirect { request_redirect } => {
+                let filter = convert::req_redirect(request_redirect)?;
+                http_route::InboundFilter::RequestRedirect(filter)
+            }
+        };
+        Ok(filter)
+    }
+}
+
+impl InboundParentRef {
+    fn convert_all(
+        route_ns: Option<&str>,
+        parent_refs: Option<Vec<api::ParentReference>>,
+    ) -> Result<Vec<Self>, InvalidParentRef> {
```

Comment:
This doesn't actually convert all of the parent references (and it shouldn't). It ignores parent references that are not servers.

```suggestion
    fn collect_from(
        route_ns: Option<&str>,
        parent_refs: Option<Vec<api::ParentReference>>,
    ) -> Result<Vec<Self>, InvalidParentRef> {
```